sway/ipc-i3.c

```diff
@@ -0,0 +1,89 @@
+#include <json-c/json.h>
+#include "sway/config.h"
+#include "sway/ipc-json.h"
+#include "sway/ipc-sway.h"
+#include "sway/tree/view.h"
+
+static void ipc_i3_json_describe_view(struct sway_container *c, json_object *object) {
+	const char *app_id = view_get_app_id(c->view);
+	const char *class = app_id;
+	const char *instance = app_id;
+	const char *role = NULL;
+	uint32_t parent_id = 0;
+	uint32_t window_id = 0;
+
+	(void) role;
+	(void) parent_id;
+	(void) window_id;
+#ifdef HAVE_XWAYLAND
+	if (c->view->type == SWAY_VIEW_XWAYLAND) {
+		class = view_get_class(c->view);
+		instance = view_get_instance(c->view);
+		role = view_get_window_role(c->view);
+		parent_id = view_get_x11_parent_id(c->view);
+		window_id = view_get_x11_window_id(c->view);
+	}
+#endif
+
+	json_object_object_add(object, "window", json_object_new_int(window_id));
+
+	json_object *window_props = json_object_new_object();
+
+	if (class) {
+		json_object_object_add(window_props, "class", json_object_new_string(class));
```

Comment:
Shouldn't these be NULL instead if unset?

Reply:
i3 omits all fields in `window_props`, except for `transient_for`, for some reason.